app/models/piece.rb

```diff
@@ -1,3 +1,102 @@
 class Piece < ApplicationRecord
   belongs_to :game
+  validates :color, presence: true
+  validates :type, presence: true
+
+  scope :white, ->() { where(color: 'white') }
+  scope :black, ->() { where(color: 'black') }
+
+  def self.at(x_position, y_position)
+    Piece.find_by(x_position: x_position, y_position: y_position)
+  end
+
+  def exist?(x, y)
+    Piece.where(x_position: x, y_position: y).present?
```

Comment:
Rails has a built in method you could use instead.

http://api.rubyonrails.org/v4.2.10/classes/ActiveRecord/FinderMethods.html#method-i-exists-3F

`pieces.exists?(x_position: x, y_position: y)`